tests/federation_room_join_partial_state_test.go

```diff
@@ -3719,8 +3719,61 @@ func TestPartialStateJoin(t *testing.T) {
 			// Cleanup.
 			psjResult.FinishStateRequest()
 		})
-		// TODO: tests which assert that:
-		//   - Join + remote ban works, then cannot rejoin
```

Comment:
I see this comment is being removed but it seems that the test below is testing something slightly different (as it only tests that the ban causes a leave, rather than then also testing after the successful leave-ban that she cannot rejoin)?

Reply:
Actually this was fine. (I was worried that I'd have to change the send_join response, but if I do it before the resync I can test that the joining HS still enforces the ban.)